doc/HOWTO_RELEASE.rst.txt

```diff
@@ -327,19 +409,41 @@ For example::
 
 Update PyPi
 -----------
+
 The final release (not betas or release candidates) should be uploaded to PyPi.
 There are two ways to update PyPi, the first one is::
 
-  $ python setup.py sdist upload
+    $ git clean -fxd  # to be safe
+    $ python setup.py sdist --formats=gztar,zip  # to check
+    # python setup.py sdist --formats=gztar,zip upload --sign
 
-and the second one is to upload the PKG_INFO file inside the sdist dir in the
+This will ask for your key PGP passphrase, in order to sign the built source
+packages.
+
+The second way is to upload the PKG_INFO file inside the sdist dir in the
 web interface of PyPi. The source tarball can also be uploaded through this
-interface. A simple binary installer for windows, created with
-``bdist_wininst``, should also be uploaded to PyPi so ``easy_install numpy``
-works.
+interface.
+
+To push the travis-ci OSX wheels up to pypi see :
+https://github.com/MacPython/numpy-wheels#uploading-the-built-wheels-to-pypi
+
+.. _push-tag-and-commit:
+
+Push the release tag and commit
+-------------------------------
+
+Finally, now you are confident this tag correctly defines the source code that
+you released you can push the tag and release commit up to github::
+
+    git push  # Push release commit
+    git push origin <version>  # Push tag named <version>
```

Comment:
Can you change "origin" to "upstream"? That's how we use it in http://docs.scipy.org/doc/numpy-dev/dev/gitwash/development_workflow.html#development-workflow, and most people will have "origin" point to their personal fork on Github.